python/visualization.py

```diff
@@ -336,11 +347,15 @@ def sort_points(xy):
 
 def plot_eps(sim, ax, output_plane=None, eps_parameters=None, frequency=None):
     # consolidate plotting parameters
-    eps_parameters = default_eps_parameters if eps_parameters is None else dict(default_eps_parameters, **eps_parameters)
+    if eps_parameters is None:
+        eps_parameters = default_eps_parameters
+    else:
+        eps_parameters = dict(default_eps_parameters, **eps_parameters)
 
     # Determine a frequency to plot all epsilon
     if frequency is not None:
-        warnings.warn('The frequency parameter of plot2D has been deprecated. Use the frequency key of the eps_parameters dictionary instead.')
+        warnings.warn("The frequency parameter of plot2D has been deprecated. "
+                      "Use the frequency key of the eps_parameters dictionary instead.")
```

Comment:
I really wish you wouldn't mix bugfix PRs with a bunch of unrelated changes.  It's a bad habit.